lib/matplotlib/image.py

```diff
@@ -661,7 +661,8 @@ def contains(self, mouseevent):
         # collection on nonlinear transformed coordinates.
         # TODO: consider returning image coordinates (shouldn't
         # be too difficult given that the image is rectilinear
-        x, y = mouseevent.xdata, mouseevent.ydata
```

Comment:
Since you aren't using `xdata`/`ydata` any more, does that mean the comment about about `figimage` is fixed?

Reply:
I believe it should be fine to remove the `self.axes is not mouseevent.inaxes` check now and `contains()` should also work for FigureImage. However, FigureImage doesn't have a `get_cursor_data()` yet, so I think the only place where `contains` would be used for a FigureImage is for the pick events? I'm not sure about that though, maybe @anntzer can help since he wrote the original comment.

Reply:
It's likely that this fixes some of the problems I had noticed when I put in the comment, but I haven't reviewed this carefully enough to see whether all problems are indeed fixed.